keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1524 @@
+    },
+    "y064": {
+        "depths": [2, 7, 14, 2],
+        "widths": [144, 288, 576, 1296],
+        "group_width": 72,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y080": {
+        "depths": [2, 4, 10, 1],
+        "widths": [168, 448, 896, 2016],
+        "group_width": 56,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y120": {
+        "depths": [2, 5, 11, 1],
+        "widths": [224, 448, 896, 2240],
+        "group_width": 112,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y160": {
+        "depths": [2, 4, 11, 1],
+        "widths": [224, 448, 1232, 3024],
+        "group_width": 112,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+    "y320": {
+        "depths": [2, 5, 12, 1],
+        "widths": [232, 696, 1392, 3712],
+        "group_width": 232,
+        "default_size": 224,
+        "block_type": "Y",
+    },
+}
+
+BASE_DOCSTRING = """Instantiates the {name} architecture.
+
+  Reference:
+    - [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+    (CVPR 2020)
+
+  For image classification use cases, see
+  [this page for detailed examples](
+  https://keras.io/api/applications/#usage-examples-for-image-classification-models).
+
+  For transfer learning use cases, make sure to read the
+  [guide to transfer learning & fine-tuning](
+    https://keras.io/guides/transfer_learning/).
+
+
+  The naming of models is as follows: `RegNet<block_type><flops>` where
+  `block_type` is one of `(X, Y)` and `flops` signifies hundred million
+  floating point operations. For example RegNetY064 corresponds to RegNet with
+  Y block and 6.4 giga flops (64 hundred million flops).
+
+  Args:
+    include_rescaling: whether or not to Rescale the inputs.If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+    include_top: Whether to include the fully-connected
+        layer at the top of the network.
+    weights: One of `None` (random initialization), or the path to the weights
+          file to be loaded. Defaults to `None`.
+    input_shape: Optional shape tuple, only to be specified
+        if `include_top` is False.
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    num_classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
```

Comment:
Let's move this (in the API and in the docstring) to be  directly below `include_top` (for consistency with other models, and because they are strongly related)